vars/promoteArtifacts.groovy

```diff
@@ -31,10 +31,16 @@ void call(Map args = [:]) {
     argsMap = [:]
     argsMap['signatureType'] = '.sig'
 
-    //////////// Signing Artifacts
-    println("Signing Core Pluings")
-    String corePluginDir = "$WORKSPACE/artifacts/$artifactPath/builds/$filename/core-plugins"
-    argsMap['artifactPath'] = corePluginDir
+
+    if(filename == "opensearch") {
```

Comment:
This `if` points to an anti-pattern where we do different things depending on the product name. I see a few options:

- check whether `core-plugins` exists, then set `artifactsPath` accordingly
- add `promoteArtifactsOpenSearch` that calls `promoteArtifacts` for the common parts, or just have multiple files

Reply:
switched to "check whether core-plugins exists"